app/src/shared/hooks/useLastMessageTimestamp.js

```diff
@@ -0,0 +1,94 @@
+import { useEffect, useCallback, useReducer, useRef } from 'react'
+import { useClient } from 'streamr-client-react'
+import useIsMounted from '$shared/hooks/useIsMounted'
+
+const SET_TIMESTAMP = 'set timestamp'
+
+const REFRESH = 'refresh'
+
+const FAIL = 'fail'
+
+export const INVALID_TIMESTAMP = -1
+
+const initialState = {
+    refreshedAt: undefined,
+    refreshKey: 0,
+    timestamp: undefined,
+}
+
+const reducer = (state, action) => {
+    switch (action.type) {
+        case REFRESH:
+            return {
+                ...state,
+                refreshKey: state.refreshKey + 1,
+            }
+        case SET_TIMESTAMP:
+            return {
+                ...state,
+                refreshedAt: state.refreshKey ? new Date().getTime() : undefined,
+                timestamp: action.timestamp,
+            }
+        case FAIL:
+            return reducer(state, {
+                type: SET_TIMESTAMP,
+                timestamp: INVALID_TIMESTAMP,
+            })
+        default:
+            return state
+    }
+}
+
+const useLastMessageTimestamp = (streamId) => {
+    const client = useClient()
+
+    const [{ timestamp, refreshedAt, refreshKey }, dispatch] = useReducer(reducer, initialState)
+
+    const isMounted = useIsMounted()
+
+    const refresh = useCallback(() => {
+        dispatch({
+            type: REFRESH,
+        })
+    }, [])
+
+    const refreshKeyRef = useRef(refreshKey)
+
+    useEffect(() => {
+        refreshKeyRef.current = refreshKey
+
+        const attach = async () => {
+            try {
+                await client.resend({
```

Comment:
There is some weirdness here with the `client` that it keeps updating if the resend was not successful, which in turn runs the effect again. You can try it by setting the params as:

```js
await client.resend({
    stream: `invalid-${streamId}`,
    ....
})
```

The loading will go crazy:
![ezgif-2-87c10eca867a](https://user-images.githubusercontent.com/1064982/96864575-4ec00880-1471-11eb-8dfe-5ca46c6d56c2.gif)

See also https://linear.app/streamr/issue/FRONT-113/stream-client-websocket-connection-keeps-failing-after-automatic

Reply:
Haha, what a gem! Thanks @juhah for finding that one. [`FRONT-113`](https://linear.app/streamr/issue/FRONT-113/stream-client-websocket-connection-keeps-failing-after-automatic) is next on my list!